### PR TITLE
[MNG-7616] Fix missing exported artifacts

### PR DESCRIPTION
**Pull request by gnodet**
Following this checklist to help us incorporate your
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MNG-XXX] SUMMARY`, where you replace `MNG-XXX`
       and `SUMMARY` with the appropriate JIRA issue. Best practice is to use the JIRA issue
       title in the pull request title and in the first line of the commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by gnodet** [comment]
@cstamas I think a few resolver artifacts are missing, but I'm not even sure they are part of the default class loader.  Could you double check ?

File: maven-core/src/main/resources/META-INF/maven/extension.xml [hunk 1]
```diff
@@ -136,6 +136,16 @@ under the License.
...
   <exportedArtifacts>
+    <!-- maven 4 api -->
+    <exportedArtifact>org.apache.maven:maven-api-core</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-api-meta</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-api-model</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-api-settings</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-api-toolchain</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-api-xml</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-xml-impl</exportedArtifact>
+    <exportedArtifact>org.apache.maven:plexus-utils</exportedArtifact>
```
Comment (kwin) [inline maven-core/src/main/resources/META-INF/maven/extension.xml:147]: I don't think this works, as only a handful of classes are actually exposed. For that reason the old plexus-utils was also no listed here.

Comment (gnodet) [reply maven-core/src/main/resources/META-INF/maven/extension.xml]: Good point. I've also removed the `maven-xml-impl` for the same reason.

**Review by kwin — APPROVED** [review]

**Comment by cstamas** [comment]
> @cstamas I think a few resolver artifacts are missing, but I'm not even sure they are part of the default class loader. Could you double check ?

Nope, all good, connector and transport is not part of public api, they were never published

**Review by cstamas — APPROVED** [review]

**Pull request merged**